homeassistant/components/dlink/switch.py

```diff
@@ -4,80 +4,23 @@
 from datetime import timedelta
 from typing import Any
 
-import voluptuous as vol
-
-from homeassistant.components.switch import (
-    PLATFORM_SCHEMA,
-    SwitchEntity,
-    SwitchEntityDescription,
-)
-from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
-from homeassistant.const import (
-    ATTR_TEMPERATURE,
-    CONF_HOST,
-    CONF_NAME,
-    CONF_PASSWORD,
-    CONF_USERNAME,
-    UnitOfTemperature,
-)
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
 from homeassistant.core import HomeAssistant
-import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-
-from .const import (
-    ATTR_TOTAL_CONSUMPTION,
-    CONF_USE_LEGACY_PROTOCOL,
-    DEFAULT_NAME,
-    DEFAULT_USERNAME,
-    DOMAIN,
-)
+
+from .const import ATTR_TOTAL_CONSUMPTION, DOMAIN
 from .entity import DLinkEntity
 
 SCAN_INTERVAL = timedelta(minutes=2)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Required(CONF_PASSWORD, default=""): cv.string,
-        vol.Required(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_USE_LEGACY_PROTOCOL, default=False): cv.boolean,
-    }
-)
-
 SWITCH_TYPE = SwitchEntityDescription(
     key="switch",
     name="Switch",
 )
 
 
-def setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
-) -> None:
-    """Set up a D-Link Smart Plug."""
-    async_create_issue(
-        hass,
-        DOMAIN,
-        "deprecated_yaml",
-        breaks_in_ha_version="2023.4.0",
```

Comment:
Might be a bit quick to merge this change? Sure we are one more release down the line. Not sure what our policy is here.

Reply:
This should actually have been removed in april release, hence that's when the break happens.
A new config flow now to be released in .6 would break in .8

Reply:
We have a general deprecation period of 2 release cycles. This would now make it 4 already, which is double the regular time.